src/kvstore.c

```diff
@@ -61,8 +62,12 @@ struct _kvstoreDictIterator {
 
 /* Dict metadata for database, used for record the position in rehashing list. */
 typedef struct {
+    /* Internal metadata */
     listNode *rehashing_node;   /* list node in rehashing list */
-} kvstoreDictMetadata;
+    
+    /* External metadata */
+    kvstoreDictMetadata meta;
+} kvstoreDictMetadataEx;
```

Comment:
I think we still need to maintain some distinction. Since following my last commit we conditionally add the histogram to dict, only in kvstore case of "keys".